src/page-modules/assistant/layout.tsx

```diff
@@ -0,0 +1,222 @@
+  initialToFeature,
+  initialTransportModesFilter,
+}: AssistantLayoutProps) {
+  const { t } = useTranslation();
+  const router = useRouter();
+
+  const [showAlternatives, setShowAlternatives] = useState(false);
+  const [selectedFromFeature, setSelectedFromFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialFromFeature);
+  const [selectedToFeature, setSelectedToFeature] = useState<
+    GeocoderFeature | undefined
+  >(initialToFeature);
+  const [departureDate, setDepartureDate] = useState<DepartureDate>({
+    type: DepartureDateState.Now,
+  });
+  const [transportModeFilter, setTransportModeFilter] = useState(
+    getInitialTransportModeFilter(initialTransportModesFilter),
+  );
+
+  const onSwap = () => {
+    let query = {};
+    const filter = filterToQueryString(transportModeFilter);
+
+    if (filter) {
+      query = {
+        filter,
+      };
+    }
+
+    const arriveBy = departureDate.type === DepartureDateState.Arrival;
+    if (arriveBy) {
+      query = {
+        ...query,
+        arriveBy,
+      };
+    }
+
+    const when = departureDate.type !== DepartureDateState.Now;
+    if (when) {
+      query = {
+        ...query,
+        when: departureDate.dateTime,
+      };
+    }
+    const fromToQuery = featuresToFromToQuery(
+      selectedToFeature,
+      selectedFromFeature,
+    );
+
+    router.push({
+      pathname: `/assistant`,
+      query: {
+        ...query,
+        ...fromToQuery,
+      },
+    });
+  };
+
+  const onSubmitHandler: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+
+    let query = {};
+
+    const filter = filterToQueryString(transportModeFilter);
+
+    if (filter) {
+      query = {
+        filter,
+      };
+    }
+
+    const arriveBy = departureDate.type === DepartureDateState.Arrival;
+    if (arriveBy) {
+      query = {
+        ...query,
+        arriveBy,
+      };
+    }
+
+    const when = departureDate.type !== DepartureDateState.Now;
+    if (when) {
+      query = {
+        ...query,
+        when: departureDate.dateTime,
+      };
+    }
+
+    const fromToQuery = featuresToFromToQuery(
+      selectedFromFeature,
+      selectedToFeature,
+    );
+
+    router.push({
+      pathname: `/assistant`,
+      query: {
+        ...query,
+        ...fromToQuery,
+      },
+    });
+  };
+
+  return (
+    <div>
+      <form className={style.container} onSubmit={onSubmitHandler}>
+        <div className={style.main}>
+          <div className={style.input}>
+            <Typo.p textType="body__primary--bold" className={style.heading}>
+              {t(PageText.Assistant.search.input.label)}
+            </Typo.p>
+
+            <Search
+              label={t(PageText.Assistant.search.input.from)}
+              onChange={setSelectedFromFeature}
+              initialQuery={
+                initialFromFeature?.name
+                  ? `${initialFromFeature.name}, ${initialFromFeature.locality}`
+                  : ''
```

Comment:
Could also be a named function for easier ways to read

Reply:
I've made some changes now to reflect what we discussed @adriansberg . Take a look and see if it's something along the lines you were thinking. It's still an issue with updating the `inputValue`, but hopefully I'll find a solution for that soon.

Reply:
Looks like we need to either make the Search controlled, and set the state when routing, or trigger a reload of the page when routing such that the initialFeature is used. Not sure what's the best solution here.

Reply:
I've updated to use `selectedItem` so that the input is updated correctly without having to reload the page